source/agora/network/RPC.d

```diff
@@ -384,13 +385,18 @@ private class RPCConnection
       impl = The object that will handle the call
       address = The address to bind to (netmask, e.g. `0.0.0.0` for all)
       port = The port to bind to
+      isBannedDg = Delegate for checking if sending peer is banned
 
 *******************************************************************************/
 
 public TCPListener listenRPC (API) (API impl, string address, ushort port,
-    Duration timeout, void delegate (agora.api.Validator.API api) @safe nothrow discoverFromClient)
+    Duration timeout, void delegate (agora.api.Validator.API api) @safe nothrow discoverFromClient,
+    RejectConnectionPredicate isBannedDg)
 {
     auto callback = (TCPConnection stream) @safe nothrow {
+        if (isBannedDg(stream.remoteAddress))
```

Comment:
Can we log this? So that we dont pull our hair out when we cant figure out why we are dropping connections.

Reply:
Trace or Debug?